### DISCUSSION TITLE
Generating OpenAPI specification from typed api directory.

### DISCUSSION THREAD

**Discussion by tintin10q** [nuxt/nuxt#15925] [Ideas]
The `$fetch` function knows what the responses can be from an API endpoint. It would be really cool and useful if you could use the `nuxi` cli to generate a swagger openapi specification for the api endpoints.

**Comment by kogratte** [comment]
Upvote :) Exactly the discussion I was looking for. 

Adding a generated OpenApi spec may allow for clients generation. It looks very, very valuable.

**Comment by zerounix** [comment]
Would this help https://nitro.unjs.io/config#openapi ?

**Reply by tintin10q** [reply]
Probably!
